llvm/lib/TableGen/DetailedRecordsBackend.cpp

```diff
@@ -152,15 +152,14 @@ void DetailedRecordsEmitter::printTemplateArgs(const Record &Rec,
 // are enclosed in parentheses.
 void DetailedRecordsEmitter::printSuperclasses(const Record &Rec,
                                                raw_ostream &OS) {
-  ArrayRef<std::pair<const Record *, SMRange>> Superclasses =
-      Rec.getSuperClasses();
+  std::vector<const Record *> Superclasses = Rec.getSuperClasses();
   if (Superclasses.empty()) {
     OS << "  Superclasses: (none)\n";
     return;
   }
 
   OS << "  Superclasses:";
-  for (const auto &[ClassRec, Loc] : Superclasses) {
+  for (const auto *ClassRec : Superclasses) {
```

Comment:
(style nit) Here and in other places where structured bindings have been removed:
```suggestion
  for (const Record *ClassRec : Superclasses) {
```

Reply:
Agreed. Done.